### DISCUSSION TITLE
Assembly-CSharp not hooking in Android game

### DISCUSSION THREAD

**Discussion by 1GN1tE** [vfsfitvnm/frida-il2cpp-bridge#314] [General]
I can see Assembly-CSharp classes but it is not hooked by tracer. When I used `Il2Cpp.dump();` function it dumped `Assembly-CSharp` assembly classes. E.g.
```cs
// Assembly-CSharp
class Microsoft.CodeAnalysis.EmbeddedAttribute : System.Attribute
{
    System.Void .ctor(); // 0x05452358
}

// Assembly-CSharp
class System.Runtime.CompilerServices.IsReadOnlyAttribute : System.Attribute
{
    System.Void .ctor(); // 0x04f1dfcc
}
```
Now when I trace the libray it shows nothing.
My code: 
```ts
import "frida-il2cpp-bridge";

Il2Cpp.perform(() => {
    Il2Cpp.trace()
        .assemblies(Il2Cpp.domain.assembly("Assembly-CSharp"))
        .and()
        .attach("full");
});
```
Output:
```
frida -U -f com.package -l .\index.js
     ____
    / _  |   Frida 16.0.19 - A world-class dynamic instrumentation toolkit
   | (_| |
    > _  |   Commands:
   /_/ |_|       help      -> Displays the help system
   . . . .       object?   -> Display information about 'object'
   . . . .       exit/quit -> Exit
   . . . .
   . . . .   More info at https://frida.re/docs/home/
   . . . .
   . . . .   Connected to SM M127G (id=REDACTED)
Spawned `com.package`. Resuming main thread!
[SM M127G::com.package ]->
```
After playing the fame for a few minutes.
So idk if there is some anti-debug going on or not. Unity Version : 2019.4.40f1c1.

**Comment by 1GN1tE** [comment]
Update: Hooking individual class works.

**Comment by vfsfitvnm** [comment]
It looks like you are using an outdated version, would you update to the latest one?

**Reply by 1GN1tE** [reply]
I am using version 0.8.2

**Reply by vfsfitvnm** [reply]
The script is outdated then (is still works of course because not that much changed, but please re-take a look at the wiki)

**Comment by vfsfitvnm** [comment]
I think the reason are either:
1) You are tracing a huge assembly with tens of thousands of methods, it will take a while to complete the setup (how much time does it take to dump the game?);
2) A *silent* exception occurred somewhere;
3) The game has some anti-debugging technique (but it didn't crash, did it?).

PS:
If you are spawning the app, I suggest you to schedule the callback on the main thread, so that you don't miss early method calls:
```ts
Il2Cpp.perform(() => {
    // code here
}, "main");
```

With this code, and if 1) is the correct reason, you will notice the game takes a while (possibly minutes) until it loads (and spoiler: it won't be playable, so it's not a good idea to trace a whole assembly, especially if that huge).
